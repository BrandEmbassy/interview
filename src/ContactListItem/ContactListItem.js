import React, { Component } from 'react';

import EditableTextBox from '../EditableTextBox/EditableTextBox';
import EditableTextArea from '../EditableTextArea/EditableTextArea';
import EditableList from '../EditableList/EditableList';

import faceIcon from './face.svg';
import saveIcon from './save.svg';
import editIcon from './edit.svg';
import deleteIcon from './delete.svg';
import './ContactListItem.css';

class ContactListItem extends Component {
  constructor(props) {
    super(props);

    const { firstName, lastName } = this.props.contact || {};

    this.state = {
      expanded: this.props.isNew,
      editable: this.props.isNew,
      newValues: {
        firstName,
        lastName
      }
    };
  }

  render() {
    const bodyClassNames = ['ContactListItem__body'];

    if (this.state.expanded) bodyClassNames.push('ContactListItem__body_expanded');
    if (this.state.editable) bodyClassNames.push('ContactListItem__body_editable');

    const { contact = { emails: [], phones: [] }, prevContact = {} } = this.props;

    const headerTemplate = (text, attrs) =>
      <h2 {...attrs} className="ContactListItem__header">{text}</h2>;

    const header = () => {
      if (this.props.isNew) {
        return headerTemplate('New Contact');
      } else {
        const curInitial = contact.lastName[0].toUpperCase();
        const prevInitial = prevContact && prevContact.lastName[0].toUpperCase();

        if (curInitial !== prevInitial) {
          return headerTemplate(curInitial, { id: `ContactListItem__header-${curInitial}` });
        }
      }
    };

    const actions = () => {
      return (
        <div className="ContactListItem__actions">
          {this.state.editable ?
            (
              <span>
                <button
                  className="Contact__button Contact__button_icon"
                  disabled={this.props.readOnly || !this.state.newValues.firstName || !this.state.newValues.lastName}
                  onClick={this.onSaveClick}>
                  <img alt="Save contact" src={saveIcon} />
                </button> <a href="" onClick={this.onCancelSaveClick}>Cancel</a>
              </span>
            ) : (
              <span>
                <button
                  className="Contact__button Contact__button_icon"
                  disabled={this.props.readOnly}
                  onClick={this.onEditClick}>
                  <img alt="Edit contact" src={editIcon} />
                </button>
                <button
                  className="Contact__button Contact__button_icon Contact__button_danger"
                  disabled={this.props.readOnly}
                  onClick={this.onDeleteClick}>
                  <img alt="Delete contact" src={deleteIcon} />
                </button>
                {this.state.expanded &&
                  <a
                    href=""
                    onClick={this.onCloseClick}>Close</a>
                }
              </span>
            )
          }
        </div>
      );
    }

    return (
      <div className="ContactListItem">

        {header()}

        <div
          className={bodyClassNames.join(' ')}
          onClick={this.onBodyClick}
        >
          <div className="ContactListItem__name">
            <img alt={contact.lastName + ', ' + contact.firstName} src={faceIcon} />

            <EditableTextBox
              defaultValue={contact.lastName}
              required
              autoFocus
              placeholder="Last name"
              editable={this.state.editable}
              onChange={this.setContactValue('lastName')}
            />, <EditableTextBox
              defaultValue={contact.firstName}
              required
              placeholder="First name"
              editable={this.state.editable}
              onChange={this.setContactValue('firstName')}
            />
          </div>

          <div className="ContactListItem__details">

            <fieldset>
              <legend>Bio</legend>
              <EditableTextArea
                defaultValue={contact.bio}
                placeholder="Don't be shy, write something about you"
                editable={this.state.editable}
                onChange={this.setContactValue('bio')}
              />
            </fieldset>

            <fieldset>
              <legend>Phones</legend>
              <EditableList
                items={contact.phones}
                itemFields={{ header: 'type', data: 'number' }}
                placeholders={{ header: 'Type', data: 'Number' }}
                editable={this.state.editable}
                onChange={this.setContactValue('phones')}
              />
            </fieldset>

            <fieldset>
              <legend>Emails</legend>
              <EditableList
                items={contact.emails}
                itemFields={{ header: 'type', data: 'address' }}
                placeholders={{ header: 'Type', data: 'Email' }}
                editable={this.state.editable}
                onChange={this.setContactValue('emails')}
              />
            </fieldset>

          </div>

          {actions()}

        </div>
      </div >
    );
  }

  setContactValue = (field) => (newValue) => {
    this.setState((prevState, props) =>
      Object.assign(prevState.newValues, { [field]: newValue }));
  }

  onBodyClick = (event) => {
    if (this.state.editable) return;

    this.setState((prevState) => ({
      expanded: !prevState.expanded
    }));
  }

  onEditClick = (event) => {
    event.stopPropagation();

    this.setState((prevState) => ({
      expanded: true,
      editable: !prevState.editable
    }));
  }

  onDeleteClick = (event) => {
    event.stopPropagation();

    this.setState({
      expanded: false
    });

    this.props.handleDelete(this.props.contact);
  }

  onCloseClick = (event) => {
    event.preventDefault();
    event.stopPropagation();

    this.setState({
      expanded: false,
      editable: false
    });
  }

  onSaveClick = (event) => {
    event.preventDefault();

    if (this.props.isNew) {
      this.props.handleCreate(this.state.newValues);
    } else {
      this.props.handleUpdate(this.state.newValues, this.props.contact);
    }

    this.setState({
      editable: false
    });
  }

  onCancelSaveClick = (event) => {
    event.preventDefault();

    if (this.props.isNew) {
      this.props.handleCreate(null);
    }

    this.setState((prevState) => ({
      editable: false
    }));
  }
};

export default ContactListItem;