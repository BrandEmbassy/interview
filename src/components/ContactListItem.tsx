/// <reference path="../../node_modules/typed-react/typed-react.d.ts" />
/// <reference path="../../typings/react/react.d.ts" />
import TypedReact = require('typed-react')
import React = require('react')

interface ContactListItemProps {
    name: string
}

class ContactListItem extends TypedReact.Component<ContactListItemProps, void> {

    public render() {
        return (
            <div className="item">
                <div className="in">
                    <div className="profile-pic"></div>
                    {this.props.name}
                </div>
            </div>
        )
    }   

}

export = TypedReact.createClass(ContactListItem)