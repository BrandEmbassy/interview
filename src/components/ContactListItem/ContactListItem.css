.ContactListItem {
  margin: auto;
}

.ContactListItem__header {
  padding: 16px;
  margin: 16px 0 0 0;
  text-transform: uppercase;
  font-size: 2rem;
}

.ContactListItem__body {
  box-sizing: border-box;  
  position: relative;
  background-color: white;
  padding: 16px;
  border-bottom: 1px solid silver;  
  box-shadow: 2px 2px 4px #888888;
}

.ContactListItem__body_expanded {  
  width: 864px;
  font-size: 1.2rem;
  margin-left: -48px;
  margin-top: 16px;
  margin-bottom: 16px;
  box-shadow: 4px 4px 8px #888888;
  z-index: 2;
  animation-name: ContactListItem_anim__expand;
  animation-duration: 0.4s;  
}

@keyframes ContactListItem_anim__expand {
  from {
    margin-left: 0px;
    width: 768px;
    font-size: 1rem;    
    box-shadow: 2px 2px 4px #888888;
    opacity: 0.6;
  }
  to {
    margin-left: -48px;
    font-size: 1.2rem;
    box-shadow: 4px 4px 8px #888888;
    width: 864px;
    opacity: 1;
  }
}

.ContactListItem__body_expanded .ContactListItem__name {
  font-size: 2rem;
  margin-bottom: 32px;
}

.ContactListItem__name .EditableTextBox {
  width: calc(50% - 84px);
}

.ContactListItem__name > img {
  width: 32px;
  margin-right: 16px;
  vertical-align: middle;
}

.ContactListItem__body_expanded .ContactListItem__name > img {
  width: 120px;
  height: 120px;
}

.ContactListItem__details {
  display: none;
}

.ContactListItem__details > fieldset {
  margin-bottom: 16px;
  padding: 16px;
  border: 0;
}

.ContactListItem__details > fieldset > legend {
  font-weight: bold;
  font-size: 0.8rem;
}

.ContactListItem__details .EditableTextArea {
  width: 100%;
  font-family: inherit;
}

.ContactListItem__body_expanded .ContactListItem__details {
  display: initial;
}  

.ContactListItem__actions {
  opacity: 0;
  position: absolute;
  top: 16px;
  right: 16px;
  z-index: 1;
}

.ContactListItem__actions button {
  margin-right: 8px;
}

.ContactListItem__actions button:last-child {
  margin-right: 0;
}

.ContactListItem__body:hover .ContactListItem__actions {
  opacity: 1;
}

.ContactListItem__body_editable .ContactListItem__actions {
  opacity: 1;
}

