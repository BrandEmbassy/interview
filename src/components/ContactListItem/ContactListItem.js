import React, { Component } from 'react';
import PropTypes from 'prop-types';

import EditableTextBox from '../EditableTextBox/EditableTextBox';
import EditableTextArea from '../EditableTextArea/EditableTextArea';
import EditableList from '../EditableList/EditableList';

import { ContactType } from '../../types';

import faceIcon from './face.svg';
import saveIcon from './save.svg';
import editIcon from './edit.svg';
import deleteIcon from './delete.svg';
import './ContactListItem.css';

class ContactListItem extends Component {
  constructor(props) {
    super(props);

    this.state = {
      expanded: this.props.isNew,
      editable: this.props.isNew,
      newValues: { ...this.props.contact },
    };
  }

  render() {
    const bodyClassNames = ['ContactListItem__body'];

    if (this.state.expanded) bodyClassNames.push('ContactListItem__body_expanded');
    if (this.state.editable) bodyClassNames.push('ContactListItem__body_editable');

    const headerTemplate = (text, attrs) =>
      <h2 {...attrs} className="ContactListItem__header">{text}</h2>;

    const { contact, prevContact } = this.props;

    const header = () => {
      if (this.props.isNew) {
        return headerTemplate('New Contact');
      } else {
        const curInitial = contact.lastName[0].toUpperCase();
        const prevInitial = prevContact && prevContact.lastName[0].toUpperCase();

        if (curInitial !== prevInitial) {
          return headerTemplate(curInitial, { id: `ContactListItem__header-${curInitial}` });
        }
      }
    };

    const actions = () => {
      return (
        <div className="ContactListItem__actions">
          {this.state.editable ?
            (
              <span>
                <button
                  className="Contact__button Contact__button_icon Contact__save-button"
                  disabled={this.props.isReadOnly || !this.state.newValues.firstName || !this.state.newValues.lastName}
                  onClick={this.onSaveClick}>
                  <img alt="Save contact" src={saveIcon} />
                </button> <a
                  className="Contact__cancel-button"
                  href=""
                  onClick={this.onCancelSaveClick}
                >Cancel</a>
              </span>
            ) : (
              <span>
                <button
                  className="Contact__button Contact__button_icon"
                  disabled={this.props.isReadOnly}
                  onClick={this.onEditClick}>
                  <img alt="Edit contact" src={editIcon} />
                </button>
                <button
                  className="Contact__button Contact__button_icon Contact__button_danger"
                  disabled={this.props.isReadOnly}
                  onClick={this.onDeleteClick}>
                  <img alt="Delete contact" src={deleteIcon} />
                </button>
                {this.state.expanded &&
                  <a
                    href=""
                    onClick={this.onCloseClick}>Close</a>
                }
              </span>
            )
          }
        </div>
      );
    }

    return (
      <div className="ContactListItem">

        {header()}

        <div
          className={bodyClassNames.join(' ')}
          onClick={this.onBodyClick}
        >
          <div className="ContactListItem__name">
            <img alt={contact.lastName + ', ' + contact.firstName} src={faceIcon} />

            <EditableTextBox
              defaultValue={contact.lastName}
              required
              autoFocus
              placeholder="Last name"
              editable={this.state.editable}
              onChange={this.setContactValue('lastName')}
            />, <EditableTextBox
              defaultValue={contact.firstName}
              required
              placeholder="First name"
              editable={this.state.editable}
              onChange={this.setContactValue('firstName')}
            />
          </div>

          <div className="ContactListItem__details">

            <fieldset>
              <legend>Bio</legend>
              <EditableTextArea
                defaultValue={contact.bio}
                placeholder="Don't be shy, write something about you"
                editable={this.state.editable}
                onChange={this.setContactValue('bio')}
              />
            </fieldset>

            <fieldset>
              <legend>Phones</legend>
              <EditableList
                items={contact.phones}
                itemFields={{ header: 'type', data: 'number' }}
                placeholders={{ header: 'Type', data: 'Number' }}
                editable={this.state.editable}
                onChange={this.setContactValue('phones')}
              />
            </fieldset>

            <fieldset>
              <legend>Emails</legend>
              <EditableList
                items={contact.emails}
                itemFields={{ header: 'type', data: 'address' }}
                placeholders={{ header: 'Type', data: 'Email' }}
                editable={this.state.editable}
                onChange={this.setContactValue('emails')}
              />
            </fieldset>

          </div>

          {actions()}

        </div>
      </div >
    );
  }

  setContactValue = (field) => (newValue) => {
    this.setState((prevState, props) =>
      Object.assign(prevState.newValues, { [field]: newValue }));
  }

  onBodyClick = (event) => {
    if (this.state.editable) return;

    this.setState((prevState) => ({
      expanded: !prevState.expanded
    }));
  }

  onEditClick = (event) => {
    event.stopPropagation();

    this.setState((prevState) => ({
      expanded: true,
      editable: !prevState.editable
    }));
  }

  onDeleteClick = (event) => {
    event.stopPropagation();

    this.setState({
      expanded: false
    });

    this.props.handleDelete(this.props.contact);
  }

  onCloseClick = (event) => {
    event.preventDefault();
    event.stopPropagation();

    this.setState({
      expanded: false,
      editable: false
    });
  }

  onSaveClick = (event) => {
    event.preventDefault();

    if (this.props.isNew) {
      this.props.handleCreate(this.state.newValues);
    } else {
      this.props.handleUpdate(this.state.newValues, this.props.contact);
    }

    this.setState({
      editable: false
    });
  }

  onCancelSaveClick = (event) => {
    event.preventDefault();

    if (this.props.isNew) {
      this.props.handleCreate(null);
    }

    this.setState((prevState) => ({
      editable: false
    }));
  }
};

ContactListItem.propTypes = {
  contact: ContactType,
  prevContact: ContactType,
  isReadOnly: PropTypes.bool,
  isNew: PropTypes.bool,
  handleUpdate: PropTypes.func,
  handleDelete: PropTypes.func,
  handleCreate: PropTypes.func,
};

ContactListItem.defaultProps = {
  contact: {
    id: '',
    firstName: '',
    lastName: '',
  },
  isReadOnly: false,
  isNew: false,
};

export default ContactListItem;