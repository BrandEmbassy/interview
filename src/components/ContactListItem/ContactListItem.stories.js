import React, { Component } from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Story } from '../../../.storybook/Story';
import {
  pucciMaurizioContact,
  pucciKristynaContact,
  rossiMarioContact,
} from '../../../.storybook/sample-contacts';

import 'normalize.css';
import '../../index.css';
import '../App/App.css';

import ContactListItem from './ContactListItem';

storiesOf('ContactListItem', module)
  .add('contact', () => (
    <Story desc="Should show the contact and the header">
      <ContactListItem
        contact={pucciMaurizioContact}
        handleUpdate={action('handleUpdate')}
        handleDelete={action('handleDelete')}
        handleCreate={action('handleCreate')}
      />
    </Story>
  ))
  .add('prevContact: Same lastName', () => (
    <Story desc="Should show the contact but not the header">
      <ContactListItem
        contact={pucciMaurizioContact}
        prevContact={pucciKristynaContact}
        handleUpdate={action('handleUpdate')}
        handleDelete={action('handleDelete')}
        handleCreate={action('handleCreate')}
      />
    </Story>
  ))
  .add('prevContact: Different lastName', () => (
    <Story desc="Should show the contact and the header">
      <ContactListItem
        contact={pucciMaurizioContact}
        prevContact={rossiMarioContact}
        handleUpdate={action('handleUpdate')}
        handleDelete={action('handleDelete')}
        handleCreate={action('handleCreate')}
      />
    </Story>
  ))
  .add('isReadOnly', () => (
    <Story desc="Should show the action buttons disabled">
      <ContactListItem
        contact={pucciMaurizioContact}
        isReadOnly
        handleUpdate={action('handleUpdate')}
        handleDelete={action('handleDelete')}
        handleCreate={action('handleCreate')}
      />
    </Story>
  ))
  .add('isNew', () => (
    <Story desc="Should show an empty contact in editing mode with the header 'New Contact'">
      <ContactListItem
        isNew
        handleUpdate={action('handleUpdate')}
        handleDelete={action('handleDelete')}
        handleCreate={action('handleCreate')}
      />
    </Story>
  ));
