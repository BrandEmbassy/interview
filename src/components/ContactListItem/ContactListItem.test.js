import React from 'react';

import {
  pucciMaurizioContact,
  pucciKristynaContact,
  rossiMarioContact,
} from '../../../.storybook/sample-contacts';

import ContactListItem from './ContactListItem';

test('contact', () => {
  const wrapper = shallow(
    <ContactListItem
      contact={pucciMaurizioContact}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

test('prevContact: Same lastName', () => {
  const wrapper = shallow(
    <ContactListItem
      contact={pucciMaurizioContact}
      prevContact={pucciKristynaContact}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

test('prevContact: Different lastName', () => {
  const wrapper = shallow(
    <ContactListItem
      contact={pucciMaurizioContact}
      prevContact={rossiMarioContact}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

test('isReadOnly', () => {
  const wrapper = shallow(
    <ContactListItem
      contact={pucciMaurizioContact}
      isReadOnly
    />
  );

  expect(wrapper).toMatchSnapshot();
});

test('isNew', () => {
  const wrapper = shallow(
    <ContactListItem
      isNew
    />
  );

  expect(wrapper).toMatchSnapshot();
});

test('handleCreate: On save should call it with a contact arg', () => {
  const onHandleCreate = jest.fn();

  const wrapper = shallow(
    <ContactListItem
      isNew
      handleCreate={onHandleCreate}
    />
  );

  wrapper.setState({
    newValues: {
      firstName: 'Maurizio',
      lastName: 'Pucci',
    }
  });

  const event = { preventDefault: jest.fn() };

  wrapper.find('.Contact__save-button').simulate('click', event);

  expect(onHandleCreate.mock.calls.length).toBe(1);
  expect(onHandleCreate.mock.calls[0][0]).toEqual({
    firstName: 'Maurizio',
    lastName: 'Pucci',
  });
  expect(event.preventDefault.mock.calls.length).toBe(1);
});

test('handleCreate: On cancel should call it with a null arg', () => {
  const onHandleCreate = jest.fn();

  const wrapper = shallow(
    <ContactListItem
      isNew
      handleCreate={onHandleCreate}
    />
  );

  const event = { preventDefault: jest.fn() };

  wrapper.find('.Contact__cancel-button').simulate('click', event);

  expect(onHandleCreate.mock.calls.length).toBe(1);
  expect(onHandleCreate.mock.calls[0][0]).toBeNull();
  expect(event.preventDefault.mock.calls.length).toBe(1);
});
