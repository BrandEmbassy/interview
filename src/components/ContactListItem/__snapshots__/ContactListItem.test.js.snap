// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact 1`] = `
<div
  className="ContactListItem"
>
  <h2
    className="ContactListItem__header"
    id="ContactListItem__header-P"
  >
    P
  </h2>
  <div
    className="ContactListItem__body"
    onClick={[Function]}
  >
    <div
      className="ContactListItem__name"
    >
      <img
        alt="Pucci, Maurizio"
        src="face.svg"
      />
      <EditableTextBox
        autoFocus={true}
        defaultValue="Pucci"
        editable={false}
        onChange={[Function]}
        placeholder="Last name"
        required={true}
      />
      , 
      <EditableTextBox
        defaultValue="Maurizio"
        editable={false}
        onChange={[Function]}
        placeholder="First name"
        required={true}
      />
    </div>
    <div
      className="ContactListItem__details"
    >
      <fieldset>
        <legend>
          Bio
        </legend>
        <EditableTextArea
          defaultValue="No comment!"
          editable={false}
          onChange={[Function]}
          placeholder="Don't be shy, write something about you"
        />
      </fieldset>
      <fieldset>
        <legend>
          Phones
        </legend>
        <EditableList
          editable={false}
          itemFields={
            Object {
              "data": "number",
              "header": "type",
            }
          }
          items={
            Array [
              Object {
                "number": "100 000 000",
                "type": "casa",
              },
            ]
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Number",
              "header": "Type",
            }
          }
        />
      </fieldset>
      <fieldset>
        <legend>
          Emails
        </legend>
        <EditableList
          editable={false}
          itemFields={
            Object {
              "data": "address",
              "header": "type",
            }
          }
          items={
            Array [
              Object {
                "address": "maurizio.pucci@abc.com",
                "type": "lavoro",
              },
              Object {
                "address": "mao@xyz.io",
                "type": "privato",
              },
            ]
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Email",
              "header": "Type",
            }
          }
        />
      </fieldset>
    </div>
    <div
      className="ContactListItem__actions"
    >
      <span>
        <button
          className="Contact__button Contact__button_icon"
          disabled={false}
          onClick={[Function]}
        >
          <img
            alt="Edit contact"
            src="edit.svg"
          />
        </button>
        <button
          className="Contact__button Contact__button_icon Contact__button_danger"
          disabled={false}
          onClick={[Function]}
        >
          <img
            alt="Delete contact"
            src="delete.svg"
          />
        </button>
      </span>
    </div>
  </div>
</div>
`;

exports[`isNew 1`] = `
<div
  className="ContactListItem"
>
  <h2
    className="ContactListItem__header"
  >
    New Contact
  </h2>
  <div
    className="ContactListItem__body ContactListItem__body_expanded ContactListItem__body_editable"
    onClick={[Function]}
  >
    <div
      className="ContactListItem__name"
    >
      <img
        alt=", "
        src="face.svg"
      />
      <EditableTextBox
        autoFocus={true}
        defaultValue=""
        editable={true}
        onChange={[Function]}
        placeholder="Last name"
        required={true}
      />
      , 
      <EditableTextBox
        defaultValue=""
        editable={true}
        onChange={[Function]}
        placeholder="First name"
        required={true}
      />
    </div>
    <div
      className="ContactListItem__details"
    >
      <fieldset>
        <legend>
          Bio
        </legend>
        <EditableTextArea
          editable={true}
          onChange={[Function]}
          placeholder="Don't be shy, write something about you"
        />
      </fieldset>
      <fieldset>
        <legend>
          Phones
        </legend>
        <EditableList
          editable={true}
          itemFields={
            Object {
              "data": "number",
              "header": "type",
            }
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Number",
              "header": "Type",
            }
          }
        />
      </fieldset>
      <fieldset>
        <legend>
          Emails
        </legend>
        <EditableList
          editable={true}
          itemFields={
            Object {
              "data": "address",
              "header": "type",
            }
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Email",
              "header": "Type",
            }
          }
        />
      </fieldset>
    </div>
    <div
      className="ContactListItem__actions"
    >
      <span>
        <button
          className="Contact__button Contact__button_icon Contact__save-button"
          disabled={true}
          onClick={[Function]}
        >
          <img
            alt="Save contact"
            src="save.svg"
          />
        </button>
         
        <a
          className="Contact__cancel-button"
          href=""
          onClick={[Function]}
        >
          Cancel
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`isReadOnly 1`] = `
<div
  className="ContactListItem"
>
  <h2
    className="ContactListItem__header"
    id="ContactListItem__header-P"
  >
    P
  </h2>
  <div
    className="ContactListItem__body"
    onClick={[Function]}
  >
    <div
      className="ContactListItem__name"
    >
      <img
        alt="Pucci, Maurizio"
        src="face.svg"
      />
      <EditableTextBox
        autoFocus={true}
        defaultValue="Pucci"
        editable={false}
        onChange={[Function]}
        placeholder="Last name"
        required={true}
      />
      , 
      <EditableTextBox
        defaultValue="Maurizio"
        editable={false}
        onChange={[Function]}
        placeholder="First name"
        required={true}
      />
    </div>
    <div
      className="ContactListItem__details"
    >
      <fieldset>
        <legend>
          Bio
        </legend>
        <EditableTextArea
          defaultValue="No comment!"
          editable={false}
          onChange={[Function]}
          placeholder="Don't be shy, write something about you"
        />
      </fieldset>
      <fieldset>
        <legend>
          Phones
        </legend>
        <EditableList
          editable={false}
          itemFields={
            Object {
              "data": "number",
              "header": "type",
            }
          }
          items={
            Array [
              Object {
                "number": "100 000 000",
                "type": "casa",
              },
            ]
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Number",
              "header": "Type",
            }
          }
        />
      </fieldset>
      <fieldset>
        <legend>
          Emails
        </legend>
        <EditableList
          editable={false}
          itemFields={
            Object {
              "data": "address",
              "header": "type",
            }
          }
          items={
            Array [
              Object {
                "address": "maurizio.pucci@abc.com",
                "type": "lavoro",
              },
              Object {
                "address": "mao@xyz.io",
                "type": "privato",
              },
            ]
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Email",
              "header": "Type",
            }
          }
        />
      </fieldset>
    </div>
    <div
      className="ContactListItem__actions"
    >
      <span>
        <button
          className="Contact__button Contact__button_icon"
          disabled={true}
          onClick={[Function]}
        >
          <img
            alt="Edit contact"
            src="edit.svg"
          />
        </button>
        <button
          className="Contact__button Contact__button_icon Contact__button_danger"
          disabled={true}
          onClick={[Function]}
        >
          <img
            alt="Delete contact"
            src="delete.svg"
          />
        </button>
      </span>
    </div>
  </div>
</div>
`;

exports[`prevContact: Different lastName 1`] = `
<div
  className="ContactListItem"
>
  <h2
    className="ContactListItem__header"
    id="ContactListItem__header-P"
  >
    P
  </h2>
  <div
    className="ContactListItem__body"
    onClick={[Function]}
  >
    <div
      className="ContactListItem__name"
    >
      <img
        alt="Pucci, Maurizio"
        src="face.svg"
      />
      <EditableTextBox
        autoFocus={true}
        defaultValue="Pucci"
        editable={false}
        onChange={[Function]}
        placeholder="Last name"
        required={true}
      />
      , 
      <EditableTextBox
        defaultValue="Maurizio"
        editable={false}
        onChange={[Function]}
        placeholder="First name"
        required={true}
      />
    </div>
    <div
      className="ContactListItem__details"
    >
      <fieldset>
        <legend>
          Bio
        </legend>
        <EditableTextArea
          defaultValue="No comment!"
          editable={false}
          onChange={[Function]}
          placeholder="Don't be shy, write something about you"
        />
      </fieldset>
      <fieldset>
        <legend>
          Phones
        </legend>
        <EditableList
          editable={false}
          itemFields={
            Object {
              "data": "number",
              "header": "type",
            }
          }
          items={
            Array [
              Object {
                "number": "100 000 000",
                "type": "casa",
              },
            ]
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Number",
              "header": "Type",
            }
          }
        />
      </fieldset>
      <fieldset>
        <legend>
          Emails
        </legend>
        <EditableList
          editable={false}
          itemFields={
            Object {
              "data": "address",
              "header": "type",
            }
          }
          items={
            Array [
              Object {
                "address": "maurizio.pucci@abc.com",
                "type": "lavoro",
              },
              Object {
                "address": "mao@xyz.io",
                "type": "privato",
              },
            ]
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Email",
              "header": "Type",
            }
          }
        />
      </fieldset>
    </div>
    <div
      className="ContactListItem__actions"
    >
      <span>
        <button
          className="Contact__button Contact__button_icon"
          disabled={false}
          onClick={[Function]}
        >
          <img
            alt="Edit contact"
            src="edit.svg"
          />
        </button>
        <button
          className="Contact__button Contact__button_icon Contact__button_danger"
          disabled={false}
          onClick={[Function]}
        >
          <img
            alt="Delete contact"
            src="delete.svg"
          />
        </button>
      </span>
    </div>
  </div>
</div>
`;

exports[`prevContact: Same lastName 1`] = `
<div
  className="ContactListItem"
>
  <div
    className="ContactListItem__body"
    onClick={[Function]}
  >
    <div
      className="ContactListItem__name"
    >
      <img
        alt="Pucci, Maurizio"
        src="face.svg"
      />
      <EditableTextBox
        autoFocus={true}
        defaultValue="Pucci"
        editable={false}
        onChange={[Function]}
        placeholder="Last name"
        required={true}
      />
      , 
      <EditableTextBox
        defaultValue="Maurizio"
        editable={false}
        onChange={[Function]}
        placeholder="First name"
        required={true}
      />
    </div>
    <div
      className="ContactListItem__details"
    >
      <fieldset>
        <legend>
          Bio
        </legend>
        <EditableTextArea
          defaultValue="No comment!"
          editable={false}
          onChange={[Function]}
          placeholder="Don't be shy, write something about you"
        />
      </fieldset>
      <fieldset>
        <legend>
          Phones
        </legend>
        <EditableList
          editable={false}
          itemFields={
            Object {
              "data": "number",
              "header": "type",
            }
          }
          items={
            Array [
              Object {
                "number": "100 000 000",
                "type": "casa",
              },
            ]
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Number",
              "header": "Type",
            }
          }
        />
      </fieldset>
      <fieldset>
        <legend>
          Emails
        </legend>
        <EditableList
          editable={false}
          itemFields={
            Object {
              "data": "address",
              "header": "type",
            }
          }
          items={
            Array [
              Object {
                "address": "maurizio.pucci@abc.com",
                "type": "lavoro",
              },
              Object {
                "address": "mao@xyz.io",
                "type": "privato",
              },
            ]
          }
          onChange={[Function]}
          placeholders={
            Object {
              "data": "Email",
              "header": "Type",
            }
          }
        />
      </fieldset>
    </div>
    <div
      className="ContactListItem__actions"
    >
      <span>
        <button
          className="Contact__button Contact__button_icon"
          disabled={false}
          onClick={[Function]}
        >
          <img
            alt="Edit contact"
            src="edit.svg"
          />
        </button>
        <button
          className="Contact__button Contact__button_icon Contact__button_danger"
          disabled={false}
          onClick={[Function]}
        >
          <img
            alt="Delete contact"
            src="delete.svg"
          />
        </button>
      </span>
    </div>
  </div>
</div>
`;
